java/src/main/java/com/google/oak/client/amd/AmdAttestedOakClient.java

```diff
@@ -0,0 +1,166 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package com.google.oak.client.amd;
+
+import static com.google.oak.remote_attestation.KeyNegotiator.EncryptorType.CLIENT;
+
+import com.google.oak.client.AeadEncryptor;
+import com.google.oak.client.Encryptor;
+import com.google.oak.client.OakClient;
+import com.google.oak.client.RpcClient;
+import com.google.oak.evidence.AmdAttestationReport;
+import com.google.oak.evidence.BasicEvidence;
+import com.google.oak.evidence.EndorsementEvidence;
+import com.google.oak.remote_attestation.KeyNegotiator;
+import com.google.oak.util.Result;
+import java.security.GeneralSecurityException;
+
+/**
+ * Placeholder implementation of an AmdAttestedOakClient. To create an instance of this class, an
+ * {@code AmdAttestationReport} must be received and verified.
+ *
+ * This implementation uses {@code byte[]} as the type of the request and response.
+ */
+public class AmdAttestedOakClient<C extends RpcClient> implements OakClient<byte[], byte[]> {
```

Comment:
I don't really have a better suggestion for a name, so I think this is fine for now, but to be pedantic: "attested client" is not really 100% accurate, as the server provides the attestation evidence and the client verifies it. It is the server state that is attested to.

Reply:
I don't like this name either, but I could not come up with anything better. I'd be happy to iterate on the name as we fill out the implementation. Added a TODO.